Split/Network/Streaming/JwtTokenParser.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  JwtTokenParser.swift
+//  Split
+//
+//  Created by Javier L. Avrudsky on 10/08/2020.
+//  Copyright © 2020 Split. All rights reserved.
+//
+
+import Foundation
+
+enum JwtTokenError: Error {
+    case tokenIsNull
+    case tokenIsInvalid
+    case unknonw
+}
+
+protocol JwtTokenParser {
+    func parse(raw: String?) throws -> JwtToken
+}
+
+struct DefaultJwtTokenParser: JwtTokenParser {
+
+    private static let kPublishersChannelMetadata = "channel-metadata:publishers"
+    private static let kPublishersCannelPrefix = "[?occupancy=metrics.publishers]"
+
+    func parse(raw: String?) throws -> JwtToken {
+        guard let raw = raw else {
+            Logger.e("Error: JWT is null.")
+            throw JwtTokenError.tokenIsNull
+        }
+
+        guard let tokenData = extractTokenData(raw: raw) else {
+            Logger.e("SSE authentication JWT payload is not valid.")
+            throw JwtTokenError.tokenIsInvalid
+        }
+
+        guard let payload = Base64Utils.decodeBase64URL(base64: tokenData) else {
+            Logger.e("Could not decode SSE authentication JWT payload.")
+            throw JwtTokenError.tokenIsInvalid
+        }
+
+        do {
+            let authToken = try Json.encodeFrom(json: payload, to: SseAuthToken.self)
+            let channelList = try Json.encodeFrom(json: authToken.channels, to: [String: [String]].self)
+            let processedChannels: [String] = channelList.compactMap {
+                if $0.value.contains(Self.kPublishersChannelMetadata) {
+                    return "\(Self.kPublishersChannelMetadata)\($0.key)"
+                }
+                return $0.key
+            }
+            return JwtToken(issuedAt: authToken.issuedAt, expirationTime: authToken.expirationTime,
```

Comment:
recheck if it is `authToken.iat` and `authToken.exp` instead of `authToken.issuedAt` and `authToken.expirationTime`.